### PR TITLE
Drifiting Fox Abnormality; Take 3.

### PR DESCRIPTION
**Pull request by Moribox**
… False Kindness " status effect, as well as all related Fox-Code.

I Messed up the PR again, had to move everything to a new one again. No changes have been made to anything other than repairing the damage my poorly done rebase caused to the PR.

## About The Pull Request
- Adds the drifting fox abnormality, as well as its currently unused " Foxbrellas " and relevant code for the "False Kindness" debuff the fox applies.

## Why It's Good For The Game
More abnormalities, also the EGO has fox ears so +25 SP to everyone who sees it. ( unless they dislike the drip then its -100 I suppose. )

## Changelog
:cl:
add: Drifting Fox abnormalitiy, HE level, Black armor / Weapon.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/simple_animal/abnormality/he/drifting_fox.dm [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+	QDEL_IN(src, 10 SECONDS)
+	..()
+
+/mob/living/simple_animal/hostile/abnormality/AttackingTarget(atom/attacked_target)
+	. = ..()
+	if(!ishuman(target))
+		return..()
+	var/mob/living/carbon/human/L = target
+	L.apply_status_effect(STATUS_EFFECT_FALSEKIND)
+	return ..()
```
Comment (LanceSmites328) [inline code/modules/mob/living/simple_animal/abnormality/he/drifting_fox.dm:92]: Still calls the original method twice, meaning it hits twice per swing.
```suggestion
/mob/living/simple_animal/hostile/abnormality/AttackingTarget(atom/attacked_target)
	if(ishuman(target))
		var/mob/living/carbon/human/H = target
		H.apply_status_effect(STATUS_EFFECT_FALSEKIND)
	return ..()
```

File: code/modules/mob/living/simple_animal/abnormality/he/drifting_fox.dm [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+/datum/status_effect/false_kindness // MAYBE the black sunder shti works this time.
+	id = "false_kindness"
+	duration = 2 SECONDS //lasts 2 seconds becuase this is for an AI that attacks fast as shit, its not meant to fuck you up with other things.
+	alert_type = /atom/movable/screen/alert/status_effect/false_kindness
+	status_type = STATUS_EFFECT_UNIQUE
```
Comment (LanceSmites328) [inline code/modules/mob/living/simple_animal/abnormality/he/drifting_fox.dm:98]: Refresh status may be better here if you want it to keep the effect up as long as it attacks and then for 2 seconds after it stops.

File: code/modules/mob/living/simple_animal/abnormality/he/drifting_fox.dm [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+	base_pixel_x = -24
+	base_pixel_y = -26
+	del_on_death = FALSE
+	maxHealth = 1000
+	health = 1000
+	rapid_melee = 3
+	move_to_delay = 2
+	damage_coeff = list( RED_DAMAGE = 0.9, WHITE_DAMAGE = 1.5, BLACK_DAMAGE = 0.5, PALE_DAMAGE = 1.5 )
+	melee_damage_lower = 5
+	melee_damage_upper = 35 // Idea taken from the old PR, have a large damage range to immitate its fucked rolls and crit chance.
```
Comment (LanceSmites328) [inline code/modules/mob/living/simple_animal/abnormality/he/drifting_fox.dm:24]: ~20-140 damage per volley is kind of insane, especially for an HE. You have to take into account that this thing makes people take 30% extra black damage, which gives this this thing an average of 80 damage per cycle which is more than NT and equivalent to CENSORED's high roll, which this thing can almost double with high rolls.

File: code/modules/mob/living/simple_animal/abnormality/he/drifting_fox.dm [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+		return
+
+//mob/living/simple_animal/hostile/abnormality/drifting_fox/Life()
+	//. = ..()
+	//if(!.) // Dead
+	//	return FALSE
+	//if(health >= 900)
+	//	var/X = pick(GLOB.department_centers)
+	//	var/turf/T = get_turf(X)
+	//	new /mob/living/simple_animal/hostile/umbrella(T)
```
Comment (LanceSmites328) [inline code/modules/mob/living/simple_animal/abnormality/he/drifting_fox.dm:129]: What's the plan with these?

File: code/modules/mob/living/simple_animal/abnormality/he/drifting_fox.dm [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+/mob/living/simple_animal/hostile/umbrella
+	name = "Umbrella"
+	desc = "A tattered and worn umbrella; The fox seems to have many to spare."
+	icon = 'ModularTegustation/Teguicons/32x32.dmi'
+	icon_state = "foxbrella"
+	icon_living = "foxbrella"
+	faction = list("hostile")
+	maxHealth = 125
+	health = 125
+	damage_coeff = list (RED_DAMAGE = 1, WHITE_DAMAGE = 0.7, BLACK_DAMAGE = 0.5, PALE_DAMAGE = 2)
```
Comment (LanceSmites328) [inline code/modules/mob/living/simple_animal/abnormality/he/drifting_fox.dm:140]:
```suggestion
	damage_coeff = list(RED_DAMAGE = 1, WHITE_DAMAGE = 0.7, BLACK_DAMAGE = 0.5, PALE_DAMAGE = 2)
```

File: code/modules/mob/living/simple_animal/abnormality/he/drifting_fox.dm [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+	if(user in pet)
+		pet -= user
+
+/mob/living/simple_animal/hostile/abnormality/drifting_fox/FailureEffect(mob/living/carbon/human/user, work_type, pe)
+	datum_reference.qliphoth_change(-1)
+
+/mob/living/simple_animal/hostile/abnormality/drifting_fox/PostWorkEffect(mob/living/carbon/human/user, work_type, pe, work_time)
+	if(get_attribute_level(user, TEMPERANCE_ATTRIBUTE) <= 40)
+		datum_reference.qliphoth_change(-1)
+	return
```
Comment (LanceSmites328) [inline code/modules/mob/living/simple_animal/abnormality/he/drifting_fox.dm:72]: Duplicate proc definition

File: code/modules/mob/living/simple_animal/abnormality/he/drifting_fox.dm [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+	attack_verb_continuous = "thwacks"
+	can_breach = TRUE
+	threat_level = HE_LEVEL
+	start_qliphoth = 2
+	work_chances = list(
+		ABNORMALITY_WORK_INSTINCT = 45,
+		ABNORMALITY_WORK_INSIGHT = 45,
+		ABNORMALITY_WORK_ATTACHMENT = list(25,30,35,40,45),
+		ABNORMALITY_WORK_REPRESSION	= 0,
+	)
```
Comment (LanceSmites328) [inline code/modules/mob/living/simple_animal/abnormality/he/drifting_fox.dm:38]:
```suggestion
	work_chances = list(
		ABNORMALITY_WORK_INSTINCT = 40,
		ABNORMALITY_WORK_INSIGHT = 40,
		ABNORMALITY_WORK_ATTACHMENT = list(15, 20, 25, 30, 35),
		ABNORMALITY_WORK_REPRESSION	= 0,
	)
```
Makes it train attachment via petting and keeps it with decent rates on the others. Rates shouldn't be super high seeing as it barely cares about anything that isn't a failure.

File: code/modules/paperwork/records/info/he.dm [hunk 8]
```diff
@@ -490,3 +490,13 @@
...
 		"When activated, the abnormality's maximum Qliphoth counter lowered, all work types became less effective, work damage increased, and the abnormality generally became more aggressive.",
 		"At 1 hour and 15 minutes the maximum Qliphoth counter reduced to 1 and the abnormality became extremely dangerous.")
+
+//Drifting Fox
+/obj/item/paper/fluff/info/he/drifting_fox
+	abno_type = /mob/living/simple_animal/hostile/abnormality/drifting_fox
+	abno_code = "T-02-173"
+	abno_info = list(
+		"It was noted that working on T-02-11-03 with a temperance level of less than 2 lowered the Qliphoth counter.",
+		"Employees who pet T-02-11-03 before a work were noted to have remarkably increased success rates with the abnormalilty when completeing attachment work.",
```
Comment (LanceSmites328) [inline code/modules/paperwork/records/info/he.dm:500]:
```suggestion
		"Employees who pet T-02-11-03 before a work were noted to have remarkably increased success rates with the Abnormality when completing Attachment work.",
```

File: code/modules/paperwork/records/info/he.dm [hunk 9]
```diff
@@ -490,3 +490,13 @@
 		"When a certain amount of time passed, the abnormality \"activated\". Each activation increased the severity of the activation effects.",
 		"When activated, the abnormality's maximum Qliphoth counter lowered, all work types became less effective, work damage increased, and the abnormality generally became more aggressive.",
 		"At 1 hour and 15 minutes the maximum Qliphoth counter reduced to 1 and the abnormality became extremely dangerous.")
+
+//Drifting Fox
+/obj/item/paper/fluff/info/he/drifting_fox
+	abno_type = /mob/living/simple_animal/hostile/abnormality/drifting_fox
+	abno_code = "T-02-173"
+	abno_info = list(
+		"It was noted that working on T-02-11-03 with a temperance level of less than 2 lowered the Qliphoth counter.",
```
Comment (LanceSmites328) [inline code/modules/paperwork/records/info/he.dm:499]:
```suggestion
		"It was noted that working on T-02-11-03 with a Temperance level of less than 2 lowered the Qliphoth counter.",
```

File: code/modules/paperwork/records/info/he.dm [hunk 10]
```diff
@@ -490,3 +490,13 @@
...
+		"Employees who pet T-02-11-03 before a work were noted to have remarkably increased success rates with the abnormalilty when completeing attachment work.",
+		"When the work result was bad, the Qliphoth counter lowered.")
```
Comment (LanceSmites328) [inline code/modules/paperwork/records/info/he.dm:501]:
```suggestion
		"When the work result was Bad, the Qliphoth counter lowered.")
```

**Review by LanceSmites328 — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/simple_animal/abnormality/he/drifting_fox.dm [hunk 11]
```diff
@@ -0,0 +1,156 @@
...
+	if(user in pet)
+		pet -= user
+
+/mob/living/simple_animal/hostile/abnormality/drifting_fox/FailureEffect(mob/living/carbon/human/user, work_type, pe)
+	datum_reference.qliphoth_change(-1)
+
+/mob/living/simple_animal/hostile/abnormality/drifting_fox/PostWorkEffect(mob/living/carbon/human/user, work_type, pe, work_time)
+	if(get_attribute_level(user, TEMPERANCE_ATTRIBUTE) <= 40)
+		datum_reference.qliphoth_change(-1)
+	return
```
Comment (LanceSmites328) [inline code/modules/mob/living/simple_animal/abnormality/he/drifting_fox.dm:72]:
```suggestion
/mob/living/simple_animal/hostile/abnormality/drifting_fox/PostWorkEffect(mob/living/carbon/human/user, work_type, pe, work_time)
	if(user in pet)
		pet -= user
	if(get_attribute_level(user, TEMPERANCE_ATTRIBUTE) <= 40)
		datum_reference.qliphoth_change(-1)
	return

/mob/living/simple_animal/hostile/abnormality/drifting_fox/FailureEffect(mob/living/carbon/human/user, work_type, pe)
	datum_reference.qliphoth_change(-1)
```

**Review by LanceSmites328 — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/simple_animal/abnormality/he/drifting_fox.dm (hunk 11 above)
Comment (Moribox) [reply code/modules/mob/living/simple_animal/abnormality/he/drifting_fox.dm:72]: Thanks, Ive been looking at this for 20 minutes trying to figure out what you meant by double proc def and had no idea and was too afraid to ask. Much appreciated.

**Review by Coxswain-Navigator — APPROVED** [review]
There don't seem to be any more major issues with this and we have a bit of a backlog with abnormalities forming, so I am going to merge this. Expect a update PR from me after the holidays.

**Review by LanceSmites328 — APPROVED** [review]
Dismissing any of my previous issues. Further issues, should they arise, will be solved in further PRs.

**Pull request merged**